webapp/src/components/NewsList/NewsList.spec.js

```diff
@@ -0,0 +1,81 @@
+import { shallowMount, mount } from "@vue/test-utils";
+import NewsList from "./NewsList.vue";
+import NewsItem from "../NewsItem/NewsItem.vue";
+
+const testItems = [
+    { id: 1, title: "VueJS", votes: 1 },
+    { id: 2, title: "TDD", votes: 4 },
+    { id: 3, title: "React", votes: 0 },
+];
+const descendingTestItems = [
+    { id: 2, title: "TDD", votes: 4 },
+    { id: 1, title: "VueJS", votes: 1 },
+    { id: 3, title: "React", votes: 0 },
+];
+const ascendingTestItems = [
+    { id: 3, title: "React", votes: 0 },
+    { id: 1, title: "VueJS", votes: 1 },
+    { id: 2, title: "TDD", votes: 4 },
+];
+
+describe("NewsList.vue", () => {
+    describe("empty", () => {
+        it("renders a message when the item list is empty", () => {
+            const wrapper = shallowMount(NewsList, {
+                data() {
```

Comment:
`data()` can be considered private to a component IMHO.
Alternatively you could introduce a prop `initialItems` and pass an initial list of items as part of `propData` here.

Again, matter of preference. Accessing/mutating `data()` does seem to be common in the Vue world.